code/modules/religion/rites.dm

```diff
@@ -102,3 +102,118 @@
+		return ..()
+	return FALSE
+
+/datum/religion_rites/fireproof/invoke_effect(mob/living/user, atom/religious_tool)
+	if(!QDELETED(chosen_clothing) && get_turf(religious_tool) == chosen_clothing.loc) //check if the same clothing is still there
+		if(istype(chosen_clothing,/obj/item/clothing/suit/hooded) || istype(chosen_clothing,/obj/item/clothing/suit/space/hardsuit ))
+			for(var/obj/item/clothing/head/integrated_helmet in chosen_clothing.contents) //check if the clothing has a hood/helmet integrated and fireproof it if there is one.
+				apply_fireproof(integrated_helmet)
+		apply_fireproof(chosen_clothing)
+		playsound(get_turf(religious_tool), 'sound/magic/fireball.ogg', 50, TRUE)
+		chosen_clothing = null //our lord and savior no longer cares about this apparel
+		return TRUE
+	chosen_clothing = null
+	to_chat(user, "<span class='warning'>The clothing that was chosen for the rite is no longer on the altar!</span>")
+	return FALSE
+
+
+/datum/religion_rites/burning_sacrifice
+	name = "Candle Fuel"
+	desc = "Sacrifice a buckled burning corpse for favor, the more burn damage the corpse has the more favor you will receive."
+	ritual_length = 20 SECONDS
+	ritual_invocations = list("To feed the fire of the one true flame ...",
+	"... to make it burn brighter ...",
+	"... so that it may consume all in its path ...",
+	"... I offer you this pitiful being ...")
+	invoke_msg = "... may it join you in the amalgamation of wax and fire, and become one in the black and white scene. "
+///the burning corpse chosen for the sacrifice of the rite
+	var/mob/living/carbon/chosen_sacrifice
+
+/datum/religion_rites/burning_sacrifice/perform_rite(mob/living/user, atom/religious_tool)
+	if(!ismovable(religious_tool))
+		to_chat(user, "<span class='warning'>This rite requires a religious device that individuals can be buckled to.</span>")
+		return FALSE
+	var/atom/movable/movable_reltool = religious_tool
+	if(!movable_reltool)
+		return FALSE
+	if(!LAZYLEN(movable_reltool.buckled_mobs))
+		to_chat(user, "<span class='warning'>Nothing is buckled to the altar!</span>")
+		return FALSE
+	for(var/corpse in movable_reltool.buckled_mobs)
+		if(!iscarbon(corpse))// only works with carbon corpse since most normal mobs can't be set on fire.
+			to_chat(user, "<span class='warning'>Only carbon lifeforms can be properly burned for the sacrifice!</span>")
+			return FALSE
+		chosen_sacrifice = corpse
+		if(chosen_sacrifice.stat != DEAD)
+			to_chat(user, "<span class='warning'>You can only sacrifice dead bodies, this one is still alive!</span>")
+			return FALSE
+		if(!chosen_sacrifice.on_fire)
+			to_chat(user, "<span class='warning'>This corpse needs to be on fire to be sacrificed!</span>")
+			return FALSE
+		return ..()
+
+/datum/religion_rites/burning_sacrifice/invoke_effect(mob/living/user, atom/movable/religious_tool)
+	for(chosen_sacrifice in religious_tool.buckled_mobs) //checks one last time if the right corpse is still buckled
```

Comment:
you dont need to loop here, if works.

that said you should check the inverse and early return.